reword existing addon maintenance document and add reasoning for it (#10)
adopted-ember-addons/program-guidelines: README.md

# Program Guidelines

Sometimes an Ember addon needs a new maintainer. In order to make this a consolidated community effort, we started Adopted Ember Addons, an org where community members can find a new home for their ember addon.

## Getting your addon adopted

- To get your addon adopted, please ping one of the adopted-ember-addons org owners (`@Melanie#1618`, `@Alon#3707`, or `@knownasilya#9990`) in the [Ember Discord](https://discord.gg/emberjs) chat in the `#adopted-ember-addons` channel and let them know what repository you'd like to transfer.
- Once they are aware, complete the Addon Transfer checklist (below).

> - After the checklist is complete, transfer the repository to them here on GitHub.

Reviewer: ```suggestion
- After the checklist is complete, transfer the repository to one of them here on GitHub.
```

- The org owner is then responsible to transfer the addon to this org (Adopted Ember Addons).

**Warning**: You cannot ever fork the repository to the same account it was transferred from. Trying to do so will break GitHub's redirect of the old URL to the new.

## Addon Transfer Checklist

These are the things that need to done to transfer your addon:

- [ ] Update the `repository` field of the `package.json` to point to the new location.
- [ ] If any demo URL or homepage is referenced, it should be updated. At the very least, file an issue on the repository so the adopter can fix.
- [ ] Ensure Release practices are documented.

> - [ ] Add at least one of the adopted-ember-addons org owners (`melsumner`, `alonbukai`, or `knownasilya`) to the maintainers of the package on `npm`. Remove other maintainers.

Reviewer: ```suggestion
- [ ] Add at least one of the adopted-ember-addons org owners (`melsumner`, `alonbukai`, or `knownasilya`) to the maintainers of the package on `npm`. Make sure they have all admin permissions. Remove other maintainers, if reasonable.
```

### Terminology

This document uses the keywords *must*, *must not*, *should*, *should not* and *may*.

*Must* and *must not* express absolute requirements. Any addon within the org that do not follow these requirements should be updated as soon as possible. A pull request that changes an addon within the org accordingly to these requirements must be merged.

*Should* and *should not* express a recommendation. There may be valid reasons to not follow the recommendation in some edge cases. If so the reasons should be documented.

*May* means that an item is truly optional. It's an individual decision for each addon within the org to implement this item or not. A given decision should not be questioned to avoid changing it back and forth. 

> > This terminology is based on [RFC 2119](https://tools.ietf.org/html/rfc2119), which is used by many specification documents.

Author: The JSON:API specification uses quotes to differentiate additional examples and explanations from main specification. I'm using a similar pattern here.

### Changelog

> All addons within the org should have a changelog. The changelog may not cover versions that were released before it was introduced.

Author: Adding a changelog wasn't a requirement nor a clear recommendation before. But I think at least a recommendation fits better than a *may* for this case. I explicitly added that it may not cover historical versions to reduce the initial work for adding a changelog.

> Changelogs are important to communicate end users what is going on with the project. See [keepachangelog.com](https://keepachangelog.com/en/1.0.0/) for additional reasoning.

The changelog should be generated and updated automatically as part of the release process. [`lerna-changelog`](https://github.com/lerna/lerna-changelog) should be used to do so.

> Lerna-changelog generates changelogs based on GitHub pull requests and their labels. Please refer to [it's documentation](https://github.com/lerna/lerna-changelog#lerna-changelog) for more details.

> lerna-changelog can be integrated with `release-it` to automatically generate the changelog as part of the release process. Robert Jackson (`@rwjblue`) provides a script to automate the setup for these tools: [create-rwjblue-release-it-setup](https://github.com/rwjblue/create-rwjblue-release-it-setup). It should be used for the setup and as a reference for recommended configuration.

Author: Before it wasn't a clear recommendation to use the script. But I think it's helpful to have a reference for the recommended configuration and this script work wells for that purpose also.

I plan to add a chapter about release process which recommends to use `release-it` later.

> Please refer to the [documentation of create-rwjblue-release-it-setup](https://github.com/rwjblue/create-rwjblue-release-it-setup#create-rwjblue-release-it-setup) for usage instructions. The script could be rerun on a repository already using lerna-changelog and release-it to update the configuration to the latest recommendations.

### GitHub labels

> GitHub issues and pull request should be labeled. It should follow the [learning team's recommendations](https://github.com/emberjs/ember.js/blob/master/CONTRIBUTING.md#issue-labeling) and [lerna-changelog](https://github.com/lerna/lerna-changelog)'s defaults.

Author: I wasn't clear what *learning team's recommendations* refers to. I only found the chapter in contributing docs of `ember-source` repo. But that refers to a third-party document. The text and link doesn't fit well together anymore. Are there any documented recommendations from the learning team that I missed? Otherwise I suggest that we reword:

```suggestion
GitHub issues and pull request should be labeled. It should follow the [rules for ember-source repository](https://github.com/emberjs/ember.js/blob/master/CONTRIBUTING.md#issue-labeling) and [lerna-changelog](https://github.com/lerna/lerna-changelog)'s defaults.
```

Reviewer 2: This was my ambiguous thought! I was thinking that the learning team as one of the ember core teams spends a lot of time thinking about how to best present topics and manage contributions. That said, it doesn't necessarily follow that they've built a taxonomy for their labels. So I intended to say "whatever ember core project that has a well defined issue labeling convention". I'd hope for us to share work since the OSS dimension to addons seems quite similar to the core projects.

### Code of conduct

> The [Ember Community Guidelines](https://emberjs.com/guidelines/) apply to all repositories within the org. It may be referenced in the repository's readme or in a `CODE_OF_CONDUCT.md`.

Author: This might be the most fundamental change to the meaning. I thought we should explicitly state that the code of conduct apply to all addons under the org regardless if it's explicitly mentioned in the repo. I also thought that it should point to ember community guidelines instead of the generic Contributor Covenant. But to be honest I'm not an expert on the topic and open to change back if this doesn't make much sense.

/cc @efx

Reviewer 2: That is a great idea to defer to the community specific expectations; I had not considered that in detail. I'm with you not being an expert on the topic 😄

The changelog should be generated and updated automatically as part of the release process. [`lerna-changelog`](https://github.com/lerna/lerna-changelog) should be used to do so.

> Lerna-changelog generates changelogs based on GitHub pull requests and their labels. Please refer to [it's documentation](https://github.com/lerna/lerna-changelog#lerna-changelog) for more details.

lerna-changelog can be integrated with `release-it` to automatically generate the changelog as part of the release process. Robert Jackson (`@rwjblue`) provides a script to automate the setup for these tools: [create-rwjblue-release-it-setup](https://github.com/rwjblue/create-rwjblue-release-it-setup). It should be used for the setup and as a reference for recommended configuration.

> Please refer to the [documentation of create-rwjblue-release-it-setup](https://github.com/rwjblue/create-rwjblue-release-it-setup#create-rwjblue-release-it-setup) for usage instructions. The script could be rerun on a repository already using lerna-changelog and release-it to update the configuration to the latest recommendations.

### GitHub labels

> GitHub issues and pull request should be labeled. It should follow the [rules for ember-source repository](https://github.com/emberjs/ember.js/blob/master/CONTRIBUTING.md#issue-labeling) for issues and [lerna-changelog](https://github.com/lerna/lerna-changelog)'s defaults for pull requests.

Author: It might be confusing to have to different sets of tags for issues and pull requests. We may want to revisit this part later. I clarified which one should be applied to which to avoid confusion. The _StandardIssueLabes_ used by ember-source repository and defaults of lerna-changelog are _not_ compatible.
